### PR TITLE
Passing client headers on connection setup

### LINKED ISSUE #130: Change method signature for GatewaySessionHandler#onConnectionOpen
Change 
`io.scalecube.services.gateway.GatewaySessionHandler#onConnectionOpen`
`onConnectionOpen(... Map<String, List<String>> headers)` -> `onConnectionOpen(... Map<String, String> headers)`.  
While it's formally correct to have multimap for headers, in realilty it's super ugly to deal with it. If _multimapness_ will be really needed then client would tokenize by himself.

### LINKED ISSUE #131: Copy message headers to corresp. client transport headers
Subj.

For http: 
In HttpGatewayClient pass headers. See constructor `io.scalecube.services.gateway.transport.http.HttpGatewayClient#HttpGatewayClient`

For websocket:
Take headrs from client transport settings and set then into underlying websocket client transport. See construcrtor `io.scalecube.services.gateway.transport.websocket.WebsocketGatewayClient#WebsocketGatewayClient`

For rsocket: 
Use HeaderCodec to encode map of headers and put them into `.setupPayload()` .

### PR DESCRIPTION
**Pull request by segabriel**
Closes #131 
Closes #130

### CODE REVIEW

File: services-gateway-client-transport/src/main/java/io/scalecube/services/gateway/transport/GatewayClientSettings.java [hunk 1]
```diff
@@ -109,6 +117,7 @@ private Builder(GatewayClientSettings originalSettings) {
       this.errorMapper = originalSettings.errorMapper;
       this.keepAliveInterval = originalSettings.keepAliveInterval;
       this.wiretap = originalSettings.wiretap;
+      this.headers = Map.copyOf(originalSettings.headers);
```
Comment (artem-v) [inline services-gateway-client-transport/src/main/java/io/scalecube/services/gateway/transport/GatewayClientSettings.java]: Plz dont use java 11 API (so far)

File: services-gateway-client-transport/src/main/java/io/scalecube/services/gateway/transport/GatewayClientSettings.java [hunk 2]
```diff
@@ -191,6 +200,11 @@ public Builder errorMapper(ServiceClientErrorMapper errorMapper) {
       return this;
     }
 
+    public Builder headers(Map<String, String> headers) {
+      this.headers = Map.copyOf(headers);
```
Comment (artem-v) [inline services-gateway-client-transport/src/main/java/io/scalecube/services/gateway/transport/GatewayClientSettings.java]: Plz dont use java 11 API (so far)

File: services-gateway-netty/src/main/java/io/scalecube/services/gateway/rsocket/RSocketGatewaySession.java [hunk 3]
```diff
@@ -40,11 +39,13 @@
   public RSocketGatewaySession(
       ServiceCall serviceCall,
       ServiceMessageCodec messageCodec,
+      Map<String, String> headers,
       BiFunction<GatewaySession, ServiceMessage, ServiceMessage> messageMapper) {
     this.serviceCall = serviceCall;
     this.messageCodec = messageCodec;
     this.messageMapper = messageMapper;
     this.sessionId = SESSION_ID_GENERATOR.incrementAndGet();
+    this.headers = Map.copyOf(headers);
```
Comment (artem-v) [inline services-gateway-netty/src/main/java/io/scalecube/services/gateway/rsocket/RSocketGatewaySession.java]: Plz dont use java 11 API (so far)

File: services-gateway-netty/src/main/java/io/scalecube/services/gateway/ws/WebsocketGatewaySession.java [hunk 4]
```diff
@@ -48,14 +48,14 @@
...
-      Map<String, List<String>> headers,
+      Map<String, String> headers,
       WebsocketInbound inbound,
       WebsocketOutbound outbound,
       GatewaySessionHandler gatewayHandler) {
     this.sessionId = sessionId;
     this.codec = codec;
 
-    this.headers = Collections.unmodifiableMap(new HashMap<>(headers));
+    this.headers = Map.copyOf(headers);
```
Comment (artem-v) [inline services-gateway-netty/src/main/java/io/scalecube/services/gateway/ws/WebsocketGatewaySession.java]: Plz dont use java 11 API (so far)

**Review by artem-v — CHANGES_REQUESTED** [review]
Plz add test on http as well.

**Comment by segabriel** [comment]
@artem-v HTTP is not implemented at all, it needs lots of work to add it and refactor existing code.
Probably better is creating an issue about HTTP.

**Comment by artem-v** [comment]
> @artem-v HTTP is not implemented at all, it needs lots of work to add it and refactor existing code.
> Probably better is creating an issue about HTTP.

Ok. Plz go ahead and create HTTP issue/story (issues/stories).

**Comment by segabriel** [comment]
@artem-v done https://github.com/scalecube/scalecube-gateway/issues/136

**Comment by artem-v** [comment]
> @artem-v done #136

Also created https://github.com/scalecube/scalecube-gateway/issues/137

**Review by artem-v — APPROVED** [review]

**Pull request merged**